src/main.rs

```diff
@@ -495,6 +497,21 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
         }
     };
 
+    // Initialize Umode.
+    Umode::init(umode_map.entry());
+    // Install umode in the current cpu.
+    // Unwrap okay: it has been called after `Umode::init()`
+    PerCpu::this_cpu().set_umode(UMODE_TASK.get().unwrap().new_cpu_state());
+
+    {
+        PerCpu::this_cpu()
+            .umode()
+            .activate()
+            .expect("Could not start U-mode")
+            .run()
+            .unwrap();
```

Comment:
Does this have any dependency on the umode executable? If so, crashing seems a bit too extreme.

Reply:
This executes a stub umode. The umode is also considered part of build -- which at the moment isn't great, see previous PR. This code is strictly linked to the basic testing of the feature. On future PRs in the queue, that will add an API and in general more functional umode, this code will be removed.

Reply:
In new update, removed the unwrap and just print the result. Again, this specific code will be removed pretty soon.